### PR TITLE
Migrate to heroku

### PR DESCRIPTION
**Pull request by jur4j**

### CODE REVIEW

File: server/config.js [hunk 1]
```diff
@@ -2,8 +2,8 @@ const transenv = require('transenv').default
 
 module.exports = transenv()(({str, bool}) => {
   return {
-    env: str('NODE_ENV', 'development'),
-    port: str('po_port'),
+    env: str('NODE_ENV'),
```
Comment (jur4j) [inline server/config.js:5]: If someone forgets to set it (especially in production), I don't want it to be `development`.

File: server/config.js [hunk 2]
```diff
@@ -2,8 +2,8 @@ const transenv = require('transenv').default
...
+    port: str('PORT', 8000),
```
Comment (jur4j) [inline server/config.js:6]: Changed because Heroku defines own `PORT`.

File: server/src/index.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ app.disable('etag')
 
 if (config.forceHttps) {
   app.use((req, res, next) => {
-    if (!req.secure) {
+    if (!req.secure && req.header('x-forwarded-proto') !== 'https') {
```
Comment (jur4j) [inline server/src/index.js:32]: `req.secure` is always `false` same as `req.protocol` is `http`.

File: README.md [hunk 4]
```diff
@@ -1,32 +1,37 @@
...
 ## Usage
 
-1. `yarn dev` run the app in development mode
-2. `yarn eslint` run eslint
-3. `yarn build && yarn start` build and serve production build
+1. `yarn dev` - run the app in development mode
+1. `yarn build && yarn start` - run the app in production mode
+1. `yarn eslint` - run eslint
+1. `yarn spj` - sort each package.json
+1. `yarn spj-check` - check if each package.json is sorted
```
Comment (paldys) [inline README.md:20]:
```suggestion
 - `yarn dev` - run the app in development mode
 - `yarn build && yarn start` - run the app in production mode
 - `yarn eslint` - run eslint
 - `yarn spj` - sort each package.json
 - `yarn spj-check` - check if each package.json is sorted
```

File: README.md [hunk 5]
```diff
@@ -1,32 +1,37 @@
...
 
 ## Setup
 
  1. Install Node.js
- 2. Install [yarn](https://yarnpkg.com).
- 3. Copy `.env.template` to `.env` in `server` directory and change your settings.
- 4. `yarn install`
+ 1. Install [yarn](https://yarnpkg.com).
+ 1. Install all dependencies - run `yarn` in root folder.
+ 1. Copy `server/.env.template` into `server/.env` and change your settings.
```
Comment (paldys) [inline README.md:10]:
```suggestion
 2. Install [yarn](https://yarnpkg.com).
 3. Install all dependencies - run `yarn` in root folder.
 4. Copy `server/.env.template` into `server/.env` and change your settings.
```

File: README.md [hunk 6]
```diff
@@ -1,32 +1,37 @@
...
 
 ## Sheets
 
 1. [create service account](https://developers.google.com/identity/protocols/OAuth2ServiceAccount#creatinganaccount) and fill info in `.env`
-2. enable Google Sheets api for that account
-3. [get sheet id](https://stackoverflow.com/a/49042689/2599892) from url and insert it into `.env`
-4. share that sheet with the email from service account (same as in `.env`)
+1. enable Google Sheets api for that account
+1. [get sheet id](https://stackoverflow.com/a/49042689/2599892) from url and insert it into `.env`
+1. share that sheet with the email from service account (same as in `.env`)
```
Comment (paldys) [inline README.md:27]:
```suggestion
2. enable Google Sheets api for that account
3. [get sheet id](https://stackoverflow.com/a/49042689/2599892) from url and insert it into `.env`
4. share that sheet with the email from service account (same as in `.env`)
```

File: server/config.js (hunk 2 above)
Comment (paldys) [reply server/config.js:6]: I don't get it. Is Heroku using `PORT` env variable? So, that should be used by pay-overview too?

Comment (jur4j) [reply server/config.js:6]: Yes, pay-overview has to use `PORT` env var as value for port.

Comment (paldys) [reply server/config.js:6]: :ok_hand:

File: server/config.js (hunk 1 above)
Comment (paldys) [reply server/config.js:5]: However, with no default value, it will fail. 

Btw.. Is this even used somewhere?

Comment (jur4j) [reply server/config.js]: Seems like no, I removed `config.env`.

**Comment by jur4j** [comment]
> I don't see any `Procfile` here. Does Heroku automatically pick-up `build` and `start` from `package.json`?

Yes (mentioned in readme).

> Other than this it looks good to me. 
> 
> Please, again add some description to your commit when squashing.

:+1:

**Pull request merged**